diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index dad8c06aac26cb8cfced31da5f35ac8e95915f15..0000000000000000000000000000000000000000
diff --git a/lib/get-mapping.js b/lib/get-mapping.js
index 187a02ed65104d8e3a058d6bfe7dc8bd7c285369..10dbf17549a6a0795e868bbc2ea6d0df661da514 100644
--- a/lib/get-mapping.js
+++ b/lib/get-mapping.js
@@ -36,13 +36,26 @@ function originalEndPositionFor(sourceMap, generatedEnd) {
     // generated file end location. Note however that this position on its
     // own is not useful because it is the position of the _start_ of the range
     // on the original file, and we want the _end_ of the range.
-    const beforeEndMapping = originalPositionTryBoth(
+    let beforeEndMapping = originalPositionTryBoth(
         sourceMap,
         generatedEnd.line,
         generatedEnd.column - 1
     );
     if (beforeEndMapping.source === null) {
-        return null;
+        for (
+            let line = generatedEnd.line;
+            line > 0 && beforeEndMapping.source === null;
+            line--
+        ) {
+            beforeEndMapping = originalPositionTryBoth(
+                sourceMap,
+                line,
+                Number.MAX_SAFE_INTEGER
+            );
+        }
+        if (beforeEndMapping.source === null) {
+            return null;
+        }
     }
 
     // Convert that original position back to a generated one, with a bump
